pkg/tlsx/auto/auto.go

```diff
@@ -24,7 +24,7 @@ func New(options *clients.Options) (*Client, error) {
 	tlsClient, tlsErr := tls.New(options)
 	ztlsClient, ztlsErr := ztls.New(options)
 	opensslClient, opensslErr := openssl.New(options)
-	if tlsErr != nil && ztlsErr != nil && (opensslErr != nil && opensslErr != openssl.ErrNotSupported) {
+	if tlsErr != nil && ztlsErr != nil && (opensslErr != nil && opensslErr != openssl.ErrNotAvailable) {
```

Comment:
Just thinking out loud, we can replace this with a new errorsutil package in https://github.com/projectdiscovery/utils with variadic `errorsutil.IsAny(err, openssl.ErrNotAvailable, ...)` which checks if the error is not nil and matches any provided one. What do you think?